osu.Game/Database/ArchiveModelManager.cs

```diff
@@ -318,10 +321,11 @@ void rollback()
                 LogForModel(item, "Beginning import...");
 
                 item.Files = archive != null ? createFileInfos(archive, Files) : new List<TFileModel>();
-                item.Hash = computeHash(item, archive);
 
                 await Populate(item, archive, cancellationToken);
 
+                item.Hash = ComputeHash(item, archive);
```

Comment:
Moving this after `Populate()` is breaking beatmap import tests, as `BeatmapManager.createBeatmapDifficulties()`, which is called by its overridden `Populate()` implementation, is relying on the hash already being there at this point to determine whether there is a need to import instead of e.g. restoring a deleted item.

Reply:
I've fixed this in the simplest way possible (locally). I think for non-legacy skins this will be handled in a better way so I'm not too concerned about code quality around this logic.